examples/edhoc-rs-no_std/build.rs

```diff
@@ -0,0 +1,31 @@
+//! This build script copies the `memory.x` file from the crate root into
+//! a directory where the linker can always find it at build time.
+//! For many projects this is optional, as the linker always searches the
+//! project root directory -- wherever `Cargo.toml` is. However, if you
+//! are using a workspace or have a more complicated build setup, this
+//! build script becomes required. Additionally, by requesting that
+//! Cargo re-run the build script whenever `memory.x` is changed,
+//! updating `memory.x` ensures a rebuild of the application with the
+//! new memory settings.
+
+use std::env;
+use std::fs::File;
+use std::io::Write;
+use std::path::PathBuf;
+
+fn main() {
```

Comment:
I am likely missing something obvious but I don't understand how is this `build.rs` script invoked during build? Does cargo automatically look for `build.rs` in its path?

Reply:
Yes, cargo automatically builds & runs build.rs!

Reply:
Here's more info: https://doc.rust-lang.org/cargo/reference/build-scripts.html